internal/metrics/listener.go

```diff
@@ -73,8 +74,13 @@ func runServer(port string, registry prometheus.Gatherer, prometheusSecret *api_
 	}
 }
 
-func writeTempFile(data []byte, name string) (*os.File, error) {
-	f, err := os.CreateTemp("", name)
+func writeToSecretsPath(data []byte, name string) (*os.File, error) {
+	_, err := os.Stat(config.DefaultSecretPath)
```

Comment:
What if the `err` is caused by sth else than "not existing path"? How could we check a specific err type?